src/redisearch_rs/query_error/src/lib.rs

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::ffi::{CStr, CString};
+use std::fmt::{Debug, Display};
+use strum::FromRepr;
+
+/// cbindgen:prefix-with-name
+/// cbindgen:rename-all=ScreamingSnakeCase
+#[derive(Clone, Copy, Default, FromRepr, PartialEq, Eq)]
+#[repr(u8)]
```

Comment:
Consider changing the repr size to be the same original size of `QueryErrorCode`, so that the `testQueryErrorStrerror` test doesn't need to be modified in the swap PR. This is due to
```
  ASSERT_STREQ(QueryError_Strerror((QueryErrorCode)9999), "Unknown status code");
```
cast no longer failing b/c of `QueryErrorCode` being too small (9999 % sizeof u8 = 15).